Update dead links in licensing-and-subscription.rst (#4920)
mattermost/docs: source/about/licensing-and-subscription.rst

.. image:: ../images/mattermost_enterprise_license.png

> System Admin access is required to apply the license. If you are not a Mattermost System Admin, contact your organization's Mattermost System Admin.

Reviewer: ```suggestion
System Admin access is required in order to apply the license. If you're not a Mattermost System Admin, contact your organization's Mattermost System Admin.
```

If you are a reseller, have a non-standard contract, or want to adjust the number of active users on your license during the license period, please contact sales@mattermost.com.

System Admins will be alerted 60 days prior to license expiry via a banner in Mattermost. Select **Renew license now** to start the renewal process in the Customer Portal. You can also dismiss the banner and renew your license at a later date via **System Console > Edition and License**.

> Once you click **Renew license now**, you are taken to the renewal page in the Customer Portal, which lists your license information and account details. These are pre-populated based on the email address associated with your existing license subscription. 

Reviewer: ```suggestion
Once you select **Renew license now**, you're taken to the renewal page in the Customer Portal, which lists your license information and account details. This is pre-populated based on the email address associated with your existing license subscription.
```

How will I know when to renew my license?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> You will be notified 60 days before your license expiry that your license is due for renewal, via a blue banner displayed at the top of your Mattermost window. This banner is only visible to System Admins.

Reviewer: ```suggestion
You will be notified 60 days prior to your license expiry that your license is due for renewal, via a blue banner displayed at the top of your Mattermost window. This banner is only visible to System Admins.
```

Reviewer: We say "prior to" as it refers specifically to the event that follows, whereas "before" is more general.

What happens to my license when I delete my account?
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> When an account is deleted, the license remains valid. When the license is close to expiring, you will need to create a new profile to purchase a new license.

Reviewer: Using "in order to" ensures that it's clear that the profile has to first be created so that a license can be purchased.
